src/main/java/seedu/address/model/task/Deadline.java

```diff
@@ -96,6 +96,33 @@ public static boolean isValidDateTime(String datetime) {
         return Pattern.matches(DATETIME_REGEX, datetime);
     }
 
+    /**
+     * Checks if the input date is an invalid leap day. Invalid leap days are days specified on 29 February in
+     * years that are not leap years.
+     *
+     * @param dateParts Array containing parts of date
+     * @return true if the String is an invalid leap day, else false
+     */
+    public static boolean isInvalidLeapDay(String[] dateParts) {
```

Comment:
this one feels a bit hard code-y.
Since according to the [behaviour](https://bugs.openjdk.org/browse/JDK-8180360) here, java automatically adjusts the date to 28 if it is not a leap year and the input is 29.
Here is one alternative that you can consider since there is a [isLeapYear()](https://docs.oracle.com/javase/8/docs/api/java/time/LocalDate.html) method based on the java documentation.

Here is the pseudocode of what you can do instead of defining a whole method for this purpose:
```
const LEAP_LAST_DAY
if (date.getDayOfMonth() == LEAP_LAST_DAY) && !date.isLeapYear()) throw error
```